CaPPMS/Data/GitHubService.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Threading.Tasks;
+using Octokit;
+namespace CaPPMS.Data
+{
+    public class GitHubService
+    {
+        private GitHubClient gitHubClient = new GitHubClient(new ProductHeaderValue("UMGCApp"));
+
+        public GitHubService()
+        {
+            gitHubClient.Credentials = new Credentials(System.Environment.GetEnvironmentVariable("GITHUB_TOKEN"));
+        }
+
+        public async Task CreateRepository(string OrganizationName, string RepoName, string Description)
+        {
+            await Task.Run(() =>
+            {
+                // Create 
+                try
+                {
+                    var repository = new NewRepository(RepoName)
+                    {
+                        AutoInit = true,
+                        Description = Description,
+                        Private = false
+                    };
+                    var newRepository = System.Threading.Tasks.Task.Run(async () => await gitHubClient.Repository.Create(OrganizationName, repository).ConfigureAwait(false)); ;
```

Comment:
Why run this in another task. If we need to await here we can probably use this task.

Reply:
Now running using the same task instead of two separate tasks.